### PR TITLE
Add Medi Token

### PR DESCRIPTION
**Pull request by ianadavies**
This PR adds the Mediconnect token to CirrusCore + Some TSLint fixes

### CODE REVIEW

File: StratisCore.UI/src/app/wallet/tokens/services/tokens.service.ts [hunk 1]
```diff
@@ -26,20 +26,22 @@ export class TokensService {
...
     return result.map(t => new SavedToken(t.ticker, t.address, null, t.name, t.decimals));
   }
 
   GetAvailableTokens(): Token[] {
-    return [
-      new Token('CG1', 'CXa9fNVXPfYL9rdqiR22NoAc9kZUfBAUCu', 'Cirrus Giveaway'),
-      ...this.defaultTokens
-    ];
+    const tokens = [];
+    tokens.push(new Token('CG1', 'CXa9fNVXPfYL9rdqiR22NoAc9kZUfBAUCu', 'Cirrus Giveaway'));
```
Comment (mrtpain) [inline StratisCore.UI/src/app/wallet/tokens/services/tokens.service.ts]: Is this needed still?

File: StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { TokensService } from '../../services/tokens.service';
 })
 @Mixin([Disposable])
 export class AddTokenComponent implements OnInit, OnDestroy, Disposable {
-
+  private subscriptions: Subscription[] = [];
```
Comment (mrtpain) [inline StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts]: Minor suggestion, change this to `private subscriptions: Subscription`.

Then add all subscriptions to it further down buy using `this.subscriptions.add(someNewSub$);`

This allows all subscriptions to be held together without having to deal with an array.

Then onDestory all you do is `this.subscriptions.unsubscribe();` maybe `if (this.subscriptions) this.subscriptions.unsubscribe()` if there are no values for whatever reason.

If you curious about this practice or others, you can read about it here https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f.

Just a suggestion, could probably make some other components logic simpler.

File: StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts [hunk 3]
```diff
@@ -37,6 +37,17 @@ export class AddTokenComponent implements OnInit, OnDestroy, Disposable {
     private activeModal: NgbActiveModal,
     private genericModalService: ModalService) {
     this.registerControls();
+    this.subscriptions.push(this.token.valueChanges.subscribe(value => {
```
Comment (mrtpain) [inline StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts]: `.push` would be changed to `.add`

File: StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts [hunk 4]
```diff
@@ -46,7 +57,8 @@ export class AddTokenComponent implements OnInit, OnDestroy, Disposable {
   ngOnInit() {
   }
 
-  ngOnDestroy() {
+  public ngOnDestroy() {
+    this.subscriptions.forEach(sub => sub.unsubscribe());
```
Comment (mrtpain) [inline StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts]: Wouldn't need forEach here, would just be `this.subscriptions.unsubscribe();`

File: StratisCore.UI/src/app/wallet/tokens/services/tokens.service.ts (hunk 1 above)
Comment (ianadavies) [reply StratisCore.UI/src/app/wallet/tokens/services/tokens.service.ts]: No, probably not, removed

File: StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts (hunk 2 above)
Comment (ianadavies) [reply StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts]: Sure, noted, more of a personal preference and consistency I think. Problem is sometimes subscriptions are made conditionally at runtime, so do you keep checking if it's null then create or add, do you just create a dummy root subscription? Will leave as is for now but will play with this in the future.

File: StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts (hunk 3 above)
Comment (ianadavies) [reply StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts]: As above

File: StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts (hunk 4 above)
Comment (ianadavies) [reply StratisCore.UI/src/app/wallet/tokens/components/add-token/add-token.component.ts]: As above

**Pull request merged**